src/network/service.rs

```diff
@@ -21,23 +21,27 @@ use network_core::{
         Node, P2pService,
     },
 };
-
-use futures::future::{self, FutureResult};
-use futures::prelude::*;
+use slog::Logger;
 
 #[derive(Clone)]
 pub struct NodeService {
     channels: Channels,
     global_state: GlobalStateR,
+    logger: Logger,
 }
 
 impl NodeService {
-    pub fn new(channels: Channels, global_state: GlobalStateR) -> Self {
+    pub fn new(channels: Channels, global_state: GlobalStateR, logger: Logger) -> Self {
```

Comment:
If `GlobalState` has a logger member, the constructor could make its own clone and add a "server" tag, no need for an extra parameter.